### PR TITLE
code.google.com is shutting down so update the dep

### PR DESCRIPTION
**Pull request by jessfraz**
ping @tianon

### CODE REVIEW

**Comment by cpuguy83** [comment]
Or review #16032 :)

File: hack/vendor.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ clone git github.com/microsoft/hcsshim 7f646aa6b26bcf90caee91e93cde4a80d0d8a83e
 clone git github.com/mistifyio/go-zfs v2.1.1
 clone git github.com/tchap/go-patricia v2.1.0
 clone git golang.org/x/net 3cffabab72adf04f8e3b01c5baf775361837b5fe https://github.com/golang/net.git
-clone hg code.google.com/p/gosqlite 74691fb6f837
+clone git github.com/mxk/go-sqlite 167da9432e1f4602e95ea67b67051cfa34412e3f
```
Comment (tiborvass) [inline hack/vendor.sh]: is there a commit that's closer to the original? there's a lot of code change.

Comment (tianon) [inline hack/vendor.sh]: As far as _time_ is concerned, https://github.com/mxk/go-sqlite/commit/1ad6fca94c61dc5cb4862c3b645653b2f58bc7fc is pretty close to what we were using before. (May 30, 2013 vs May 25, 2013; https://code.google.com/p/gosqlite/source/detail?r=74691fb6f83716190870cde1b658538dd4b18eb0)

Not sure if that actually means anything, and haven't had a chance to check the code diff yet.

Comment (jfrazelle) [inline hack/vendor.sh]: ya i dont think it does :/ i tried to see the commits to the hq repo, but they dont align, however they were pointing people to use this repo, so its semi weird...

Comment (tianon) [inline hack/vendor.sh]: Do you happen to have a link to where they suggest using this particular repo?  I can't find any suggestion like that myself. :cry:

Alternatively, we could switch over to `go-sqlite3` completely (https://github.com/docker/docker/pull/15387#issuecomment-137606204).

Comment (jfrazelle) [inline hack/vendor.sh]: https://code.google.com/p/go-sqlite/

Comment (jfrazelle) [inline hack/vendor.sh]: I'm tryinng that sha tho

Comment (tianon) [inline hack/vendor.sh]: Wow, I obviously can't read. :flushed:

Comment (tianon) [inline hack/vendor.sh]: It is really crazy that it doesn't have the proper `hg` history imported though.  I can't imagine why they wouldn't.

Comment (jfrazelle) [inline hack/vendor.sh]: right its so weird!

**Comment by jfrazelle** [comment]
the diff is so completely different idk

**Comment by jfrazelle** [comment]
maybe we should just use something else :/

**Comment by tianon** [comment]
Yeah, that's why I suggested we should consider/test mattn's go-sqlite3
that we're already importing. :+1:

**Comment by jfrazelle** [comment]
ooo yeah i will pop it out i didnt know if there is a reson we werent using it already

**Comment by tianon** [comment]
There was, but it's been shown to work successfully for packaging needs, so
I'm +1. :+1:

**Comment by jfrazelle** [comment]
cool! updated!

File: pkg/graphdb/conn_sqlite3_unix.go [hunk 2]
```diff
@@ -3,5 +3,5 @@
 package graphdb
 
 import (
-	_ "code.google.com/p/gosqlite/sqlite3" // registers sqlite
+	_ "github.com/github.com/mattn/go-sqlite3" // registers sqlite
```
Comment (tianon) [inline pkg/graphdb/conn_sqlite3_unix.go]: `github.com/github.com`

Comment (jfrazelle) [inline pkg/graphdb/conn_sqlite3_unix.go]: ah i was too slow to fix and you saw it :ashamed:

**Comment by jfrazelle** [comment]
it works!

**Comment by jfrazelle** [comment]
omg @LK4D4 will be excited we only have 1 sqlite package now :confetti_ball:

**Comment by tianon** [comment]
LGTM

**Comment by diogomonica** [comment]
LGTM

**Comment by mountkin** [comment]
There's a warning when running `make binary`.

```
---> Making bundle: binary (in bundles/1.9.0-dev/binary)
Building: bundles/1.9.0-dev/binary/docker-1.9.0-dev
# github.com/docker/docker/docker
/var/tmp/go-link-pJY0ZF/000006.o: In function `unixDlOpen':
/go/src/github.com/docker/docker/vendor/src/github.com/mattn/go-sqlite3/code/sqlite3-binding.c:29925: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
```

**Comment by jfrazelle** [comment]
It's fixed in another pr






On Sat, Sep 5, 2015 at 9:07 PM -0700, "Shijiang Wei" <notifications@github.com> wrote:












There's a warning when running make binary.

---> Making bundle: binary (in bundles/1.9.0-dev/binary)
Building: bundles/1.9.0-dev/binary/docker-1.9.0-dev
# github.com/docker/docker/docker
/var/tmp/go-link-pJY0ZF/000006.o: In function `unixDlOpen':
/go/src/github.com/docker/docker/vendor/src/github.com/mattn/go-sqlite3/code/sqlite3-binding.c:29925: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

**Comment by jfrazelle** [comment]
It is just a warning tho we do not use dlopen






On Sat, Sep 5, 2015 at 9:07 PM -0700, "Shijiang Wei" <notifications@github.com> wrote:












There's a warning when running make binary.

---> Making bundle: binary (in bundles/1.9.0-dev/binary)
Building: bundles/1.9.0-dev/binary/docker-1.9.0-dev
# github.com/docker/docker/docker
/var/tmp/go-link-pJY0ZF/000006.o: In function `unixDlOpen':
/go/src/github.com/docker/docker/vendor/src/github.com/mattn/go-sqlite3/code/sqlite3-binding.c:29925: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

**Comment by mountkin** [comment]
I see. It's #16094
Thanks.

**Pull request merged**